pkg/block/fetcher.go

```diff
@@ -469,15 +585,20 @@ type MetaFetcher struct {
 //
 // Returned error indicates a failure in fetching metadata. Returned meta can be assumed as correct, with some blocks missing.
 func (f *MetaFetcher) Fetch(ctx context.Context) (metas map[ulid.ULID]*metadata.Meta, partial map[ulid.ULID]error, err error) {
-	metas, partial, err = f.wrapped.fetch(ctx, f.metrics, f.filters, f.modifiers)
+	metas, _, partial, err = f.FetchWithMarks(ctx)
+	return metas, partial, err
+}
+
+func (f *MetaFetcher) FetchWithMarks(ctx context.Context) (metas map[ulid.ULID]*metadata.Meta, marks map[ulid.ULID]*metadata.DeletionMark, partial map[ulid.ULID]error, err error) {
```

Comment:
Fine with me. I consider this method to be temporary. I think `Filter` should return deletion marks too, but I didn't want to change it in this PR, as that would touch many other places. To avoid that I've instead introduced this method.